pan3d/explorers/slice_explorer.py

```diff
@@ -422,6 +455,29 @@ def on_scalar_change(self, varmin, varmax, **_):
         self._sbar_actor.SetLookupTable(self._slice_mapper.GetLookupTable())
         self.ctrl.view_update()
 
+    def on_axis_scale_change(self, axis, value):
+        """
+        Performs all the steps necessary when user specifies scaling along a certain axis
+        """
+        s = self.state
+        update = (
```

Comment:
```python
axis_name = ["x_scale", "y_scale", "z_scale"][axis]

if s[axis_name] == value:
  return

s[axis_name] = value

self._slice_actor.SetScale(s.x_scale, s.y_scale, s.z_scale)
self._data_actor.SetScale(s.x_scale, s.y_scale, s.z_scale)
self._outline_actor.SetScale(s.x_scale, s.y_scale, s.z_scale)
self.on_view_mode_change(s.view_mode)
```

Reply:
You could even do that 

```python
axis_names = ["x_scale", "y_scale", "z_scale"]
axis_name = axis_names[axis]

if s[axis_name] == value:
  return

s[axis_name] = value
scales = [s[n] for n in axis_names]

self._slice_actor.SetScale(*scales)
self._data_actor.SetScale(*scales)
self._outline_actor.SetScale(*scales)
self.on_view_mode_change(s.view_mode)
```

Reply:
This is a great suggestion, really simplifies the code.
I forgot in the instance when I wrote the code that the `state` is just one big dictionary